contrib/platform/src/com/sun/jna/platform/win32/Advapi32.java

```diff
@@ -284,6 +301,292 @@ boolean CreateWellKnownSid(int wellKnownSidType, PSID domainSid,
+     *         function fails, the return value is zero. For extended error
+     *         information, call GetLastError.
+     */
+    boolean SetSecurityDescriptorGroup(SECURITY_DESCRIPTOR pSecurityDescriptor, PSID pGroup, boolean bGroupDefaulted);
+
+    /**
+     * The GetSecurityDescriptorDacl function retrieves a pointer to the discretionary access control list (DACL) in
+     * a specified security descriptor.
+     * @param pSecurityDescriptor
+     *              A pointer to the SECURITY_DESCRIPTOR structure that contains the DACL. The function retrieves a pointer to it.
+     * @param bDaclPresent
+     *              A pointer to a value that indicates the presence of a DACL in the specified security descriptor. If
+     *              lpbDaclPresent is TRUE, the security descriptor contains a DACL, and the remaining output parameters in this
+     *              function receive valid values. If lpbDaclPresent is FALSE, the security descriptor does not contain a DACL,
+     *              and the remaining output parameters do not receive valid values. A value of TRUE for lpbDaclPresent does not
+     *              mean that pDacl is not NULL. That is, lpbDaclPresent can be TRUE while pDacl is NULL, meaning that a NULL
+     *              DACL is in effect. A NULL DACL implicitly allows all access to an object and is not the same as an empty DACL.
+     *              An empty DACL permits no access to an object. For information about creating a proper DACL, see Creating a DACL.
+     * @param pDacl
+     *              A pointer to a pointer to an access control list (ACL). If a DACL exists, the function sets the pointer pointed
+     *              to by pDacl to the address of the security descriptor's DACL. If a DACL does not exist, no value is stored.
+     *              If the function stores a NULL value in the pointer pointed to by pDacl, the security descriptor has a NULL DACL.
+     *              A NULL DACL implicitly allows all access to an object.
+     *              If an application expects a non-NULL DACL but encounters a NULL DACL, the application should fail securely and
+     *              not allow access.
+     * @param bDaclDefaulted
+     *              A pointer to a flag set to the value of the SE_DACL_DEFAULTED flag in the SECURITY_DESCRIPTOR_CONTROL structure
+     *              if a DACL exists for the security descriptor. If this flag is TRUE, the DACL was retrieved by a default mechanism;
+     *              if FALSE, the DACL was explicitly specified by a user.
+     * @return If the function succeeds, the return value is nonzero. If the
+     *         function fails, the return value is zero. For extended error
+     *         information, call GetLastError.
+     */
+    boolean GetSecurityDescriptorDacl(SECURITY_DESCRIPTOR pSecurityDescriptor, BOOLByReference bDaclPresent, PointerByReference pDacl, BOOLByReference bDaclDefaulted);
```

Comment:
You are using ACL in the SetSecurityDescriptorDacl you could check if WinNT#ACL could be used if it enhanced with a subclass that implements Structure.ByReference